core/domain/suggestion_domain.py

```diff
@@ -72,3 +77,73 @@ def to_dict(self):
             'change_cmd': self.change_cmd,
             'score_category': self.score_category
         }
+
+    def validate(self):
+        """Validates the suggestion object. Each subclass must implement
+        this function
+        """
+        pass
+
+
+class SuggestionEditStateContent(BaseSuggestion):
```

Comment:
@nithusha21 though this design is sound, I'm not sure how consistent this design is with the codebase. In the past, we had used such a design for creating classes for ML classifiers (now deprecated) for different types of inputs (text, code) within extensions/ and we had an algorithm_registry (like interaction_registry) to retrieve the class corresponding to the input type (equivalent to suggestion type in our case). It was more or less equivalent to interactions. @seanlip what's your opinion?

Reply:
Thanks for the suggestions! I'll add the not implemented check in the superclass and yes I did plan to use the suggestion_type_domain_class_mapping while creating domain objects from models to make an object of the appropriate sub-class.